pages/blog/websocket-part3.md

```diff
@@ -0,0 +1,605 @@
+
+Created AsyncAPI document is not very complicated as this way it will be easier to understand the generated code. Most important is for you to notice that all information about your application is expressed in the AsyncAPI document, and once you do it, options for the next steps are just endless.
+
+## Generate Code
+
+It is time now to generate some code that enables you to focus just on the business logic.
+
+### AsyncAPI Generator
+
+The AsyncAPI Initiative maintains an [AsyncAPI Generator](https://github.com/asyncapi/generator/), a tool that enables you to generate anything you want out of the AsyncAPI document. [Generator](https://www.asyncapi.com/generator) provides several features that make it much easier to provide so-called **templates**. The template is a standalone project that defines what files should be rendered by the Generator as a final output.
+
+We have many templates on our [list](https://github.com/asyncapi/generator#list-of-official-generator-templates). Try out the project by following :point_down: instructions:
+
+<GeneratorInstallation />
+
+### Generate Server and Client
+
+This article focuses on WebSocket therefore, I use our [Node.js WebSocket template](https://github.com/asyncapi/nodejs-ws-template) that is capable of generating server for WebSocket API and also a client that is aware of available channels.
+
+1. Create a new directory where you will work on the project:
+      ```bash
+      mkdir shrekapp && cd shrekapp
+      ```
+1. Trigger generation using the template:
+  ```bash
+  ag https://raw.githubusercontent.com/derberg/shrekapp-asyncapi-designed/main/asyncapi.yaml @asyncapi/nodejs-ws-template -o myapp -p server=swamp
+  ```
+1. Access generated folder and list all files from the directory. Notice that Node.js application is generated:
+  ```bash
+  cd myapp && ls
+  ```
+1. Install application dependencies
+  ```bash
+  npm i
+  ```
+1. Start the application
+  ```bash
+  npm start
+  ```
+
+That is it. The basics are done. The application is ready, and all the basic logic is there. You can already interact with the application on **localhost** on port **80**. Now you need a client able to communicate with WebSocket protocol. For now, we will not explore the generated client.
+
+1. Get **websocat** (curl-like tool for WebSocket) by following [these](https://github.com/vi/websocat#installation) instructions
+1. Connect to one of the channels and notice that the server sent you a message to respond to established connection.
+  ```bash
+  websocat ws://localhost/travel/status
+
+  Message from the server: Implement here your business logic that sends messages to a client after it connects.
+  ```
+
+Logs in the running server should also indicate a new connection with the server:
+```bash
+Listening on port 80
+/travel/status client connected.
+```
+
+## Walkthrough the Code
```

Comment:
"walkthrough" is a noun, you either do:

```suggestion
## Code Walkthrough
```

Or:

```suggestion
## Walk through the Code
```

I prefer the first one.